lib/openshift/service.rb

```diff
@@ -96,6 +96,15 @@ def node_port(user: nil, port:, cached: true, quiet: false)
       return node_port
     end
 
+    def port(user: nil, name:, cached: true, quiet: false)
+      port = nil
+      ports = self.ports(user: user, cached: cached, quiet: quiet)
+      ports.each do | p |
+        port = p['port'] if p['name'] == name
+      end
+      return port
```

Comment:
@QiaolingTang should we always get back a valid port instead of nil?  Do we need to raise an error at the end of `ports.each`, there is not a match...in case where user gave an name of the port that's incorrect.  WDYT?

Reply:
> @QiaolingTang should we always get back a valid port instead of nil?
For me, yes.

> Do we need to raise an error at the end of `ports.each`, there is not a match...in case where user gave an name of the port that's incorrect. WDYT?
I'll update it, thanks a lot.

Reply:
@pruan-rht I updated it , PTAL.

```
[1] pry(#<BushSlicer::DefaultWorld>)> 
[2] pry(#<BushSlicer::DefaultWorld>)> service('fluentd').port(name: 'metric')
waiting for operation up to 3600 seconds..
RuntimeError: Could not find port with name metric, does the name really exist?
from /Users/tangqiaoling/github.com/verification-tests/lib/openshift/service.rb:105:in `port'
[3] pry(#<BushSlicer::DefaultWorld>)> service('fluentd').port(name: 'metrics')
=> 24231
[4] pry(#<BushSlicer::DefaultWorld>)> 
```

Reply:
@QiaolingTang you can do `if port.nil?` instead of `unless port != nil`

Reply:
Done

Reply:
/lgtm